File: SPX.Text/Text.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using Microsoft.Xna.Framework;
using Microsoft.Xna.Framework.Graphics;
using SPX.Core;

namespace SPX.Text
{
    public class Text
    {
        protected string _contents;
        protected Vector2 _position = new Vector2(0,0);
        protected int _textType = TextType.Inventory;

        public Text()
        { }

        public void Reset(string contents, int x, int y)
        {
            _contents = contents;
            _position.X = x;
            _position.Y = y;
        }

        public Text(string contents,int x, int y,int type)
        {
            Reset(contents, x, y);
            _textType = type;
        }
        public virtual int Update()
        {
            return 0;
        }

        public int GetTextType()
        {
            return _textType;
        }

        protected virtual void DrawText(SpriteBatch target)
        {
        }

        public virtual void Draw()
        {
           
        }
    }
}

File: proj/src/spx/text/Text.java
package spx.text;

import spx.core.Point2;
import spx.graphics.Renderer;

public class Text {
	protected String _contents;
	protected Point2 _position = new Point2(0, 0);
	protected TextType _textType = TextType.Inventory;

	public Text() {
	}

	public void Reset(String contents, int x, int y) {
		_contents = contents;
		_position.Reset(x, y);
	}

	public Text(String contents, int x, int y, TextType type) {
		Reset(contents, x, y);
		_textType = type;
	}

	public int Update() {
		return 0;
	}

	public TextType GetTextType() {
		return _textType;
	}

	protected void DrawText(Renderer target) {
	}

	public void Draw() {

	}
}
